refactor(elements)

leviovan opened the pull request:
remove null assertion and type Any
remove @input for function

leviovan committed:
refactor(elements)
remove null assertion and type Any
remove @intut for function

leviovan committed:
refactor(elements)
add services todo

tamazyanarsen commented on src/app/component/todo/todo.component.ts:
```diff
@@ -8,44 +9,56 @@ import { Status, Todo } from 'src/app/types/todo';
   id: number = 0
   filter: Status = ''
   newTodo: string | null = null
+  todos:Todo[]=[]
+  constructor(private todoService:TodoService){
+
+  }
```
а где форматирование кода?

tamazyanarsen commented on src/app/component/todo/todo.component.ts:
```diff
@@ -8,44 +9,56 @@ import { Status, Todo } from 'src/app/types/todo';
-    this.todos = [...this.todos]
+    const indexTodo = this.todos.map(t => t.id).indexOf(index)
+    this.todos.splice(indexTodo, 1)
+     this.todoService.updateTodo([...this.todos])
```
зачем лишний пробел?

tamazyanarsen commented on src/app/component/todo/todo.component.ts:
```diff
@@ -8,44 +9,56 @@ import { Status, Todo } from 'src/app/types/todo';
+    this.todoService.updateTodo([...tempTodo])
   }
   toggleStatusTodo(id: number) {
     const newTodos = this.todos.map((todo) => {
       todo.id === id && (todo.status = todo.status === 'active' ? 'completed' : 'active')
       return todo
     })
-    this.todos = [...newTodos]
+    this.todoService.updateTodo([...newTodos])
```
работа со списком должна быть в сервисе

изменение, удаление, создание и чтение

всё через сервис

leviovan committed:
todo
refactor(element)
move logic list's in services and add new function toggleStatus

tamazyanarsen commented on src/app/component/todo/todo.component.html:
```diff
@@ -12,7 +12,7 @@ <h1 class="font-bold text-center text-[80px]">Todos</h1>
       </div>
       <div *ngIf="newTodo?.length">
-        <button (click)="addTodo()">Добавить дело</button>
+        <button (click)="addTodo(newTodo!)">Добавить дело</button>
```
мы же договаривались, что не стоит использовать `!`

leviovan replied:
Исправил, плохая привычка

tamazyanarsen commented on src/app/services/todo.service.ts:
```diff
@@ -18,4 +21,30 @@ export class TodoService {
   }
+  addTodo(newTodo: string) {
+    if (newTodo?.trim()) {
+      this.id = this.id + 1;
```
id для каждого объекта todo создается на месте
это должно быть что-то уникальное, а не просто порядковый номер

например, Date.now(), в реальных проектах используют uuid

leviovan replied:
Я знаю, что id уникальное должно быть, и создал такой способ, который при создании туду создавал уникальный номер, щас изменю на Date.now

tamazyanarsen commented on src/app/services/todo.service.ts:
```diff
@@ -18,4 +21,30 @@ export class TodoService {
   updateTodo(newState: Todo[]) {
```
метод update должен обновлять конкретный todo, а не сразу весь массив

leviovan replied:
А он даже не использовался, убрал

tamazyanarsen commented on src/app/services/todo.service.ts:
```diff
@@ -18,4 +21,30 @@ export class TodoService {
+    }
+  }
+  removeTodo(id: number) {
+    const indexTodo = this.todos.value.map(t => t.id).indexOf(id)
```
замени на `findIndex`, но лучше сделать через `filter`

оставляем все элементы, кроме того, где совпадает id

leviovan replied:
Хотел с фильтром сналало, но почему-то сделал так, поправил

leviovan committed:
refactor(element)
удалил не используемую функцию updateTodo
изменил поле id на 100% универсальное значение (Date.now)
поменял логику удаления элемента
